### PR TITLE
Implement incantation command

### PR DESCRIPTION
**Pull request by AndreasLETANTER**
### Incantation Command
This command takes no arguments, it is used to let the users level up based on the stones on the ground.

It should send a message to everyone when the incantation start, and set the new level for every user at the end of the incantation.

It should check if there are enough stones and players for the incantation, at the beginning and at the end. All the players must be freezed during the incantation.

**The command have a complete error handling.**

_It has been implemented with unit tests_

### Not planned changes
The map now start by being refilled for avoiding empty map at start of the server.
Due to the change above, some unit tests wasn't working due to modification of the inventory of the tiles, added a for loop for resetting the map to 0, and some wasn't passing due to stdout now containing the debug print of the refill_map function so i added the debug print in the unit tests.

### CODE REVIEW

**Review by Rafouzou — APPROVED** [review]
Ok

File: server/src/map_handling/initialize_map.c [hunk 1]
```diff
@@ -35,5 +35,6 @@ unsigned int initialize_map(server_database_t *server_database)
     if (initialize_tiles(server_database) == ERROR) {
         return (ERROR);
     }
+    refill_map(server_database);
```
Comment (HugoN1) [inline server/src/map_handling/initialize_map.c:38]: Is it a part of incantation command ? If not, specify this as a "not planned change" and describe it in your PR comment

File: server/src/server_handling/events_handling/teams/commands_errors/tools/is_incantation_possible.c [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+*/
+static unsigned int elevation_conditions[7][7] = {
+    {1, 1, 0, 0, 0, 0, 0},
+    {2, 1, 1, 1, 0, 0, 0},
+    {2, 2, 0, 1, 0, 2, 0},
+    {4, 1, 1, 2, 0, 1, 0},
+    {4, 1, 2, 1, 3, 0, 0},
+    {6, 1, 2, 3, 0, 1, 0},
+    {6, 2, 2, 2, 2, 2, 1}
+};
```
Comment (HugoN1) [inline server/src/server_handling/events_handling/teams/commands_errors/tools/is_incantation_possible.c:24]: 2 times the same 2D array ?

Comment (HugoN1) [reply server/src/server_handling/events_handling/teams/commands_errors/tools/is_incantation_possible.c:24]: Already existing in incantation_command.c

File: server/src/server_handling/events_handling/teams/commands_errors/tools/is_incantation_possible.c [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        if (team->clients[i]->position.x == position_x &&
+        team->clients[i]->position.y == position_y &&
+        team->clients[i]->level == level_searched &&
+        team->clients[i]->request_buffer[0][0] == '\0' &&
+        team->clients[i]->request_execution_time <= 0) {
+            nb_players_on_tile++;
+        }
+    }
+    return (nb_players_on_tile);
+}
```
Comment (HugoN1) [inline server/src/server_handling/events_handling/teams/commands_errors/tools/is_incantation_possible.c:49]: third time I see the same method name...

File: server/src/server_handling/events_handling/teams/commands_errors/tools/is_incantation_possible.c [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+/**
+ * @brief Check if the incantation is possible
+ * @param client The client that sent the request
+ * @param server_database The struct containing server infos
+ * @return true If the incantation is possible
+ * @return false If the incantation is not possible
+*/
+bool is_incantation_possible(client_t *client
```
Comment (HugoN1) [inline server/src/server_handling/events_handling/teams/commands_errors/tools/is_incantation_possible.c:81]: another is_incantation_possible method ?

File: server/tests/commands/look/look_command.c [hunk 5]
```diff
@@ -39,6 +39,13 @@ Test(look_command, level1, .init = redirect_all_std)
     server_database->teams[1]->clients[0]->orientation = NORTH;
     server_database->teams[1]->clients[0]->level = 1;
     initialize_inventory(server_database->teams[1]->clients[0]);
+    for (unsigned int i = 0; i < server_database->map->height; i++) {
```
Comment (HugoN1) [inline server/tests/commands/look/look_command.c:42]: add it to "not planned changes" in PR description

File: server/tests/commands/set/set_command.c [hunk 6]
```diff
@@ -39,6 +39,13 @@ Test(set_command, should_place_only_one_food, .init = redirect_all_std)
     server_database->teams[1]->clients[0]->orientation = NORTH;
     server_database->teams[1]->clients[0]->level = 1;
     initialize_inventory(server_database->teams[1]->clients[0]);
+    server_database->map->tiles[5][5].inventory[FOOD] = 0;
```
Comment (HugoN1) [inline server/tests/commands/set/set_command.c:42]: add it too

File: server/tests/commands/inventory/inventory_command.c [hunk 7]
```diff
@@ -40,7 +40,7 @@ Test(inventory_command, should_only_ten_food, .init = redirect_all_std)
     initialize_inventory(server_database->teams[1]->clients[0]);
     
     inventory_command(server_database->teams[1]->clients[0], server_database, NULL);
-    cr_assert_stdout_eq_str(GREEN "Client 4 inventory: [ food 10, linemate 0, deraumere 0, sibur 0, mendiane 0, phiras 0, thystame 0 ]\n" RESET);
+    cr_assert_stdout_eq_str(PURPLE "Refilling map...\n" RESET GREEN "Client 4 inventory: [ food 10, linemate 0, deraumere 0, sibur 0, mendiane 0, phiras 0, thystame 0 ]\n" RESET);
```
Comment (HugoN1) [inline server/tests/commands/inventory/inventory_command.c:43]: not planned changes

File: server/tests/map_handling/clear_map.c [hunk 8]
```diff
@@ -10,7 +10,13 @@
 #include <criterion/criterion.h>
 #include <criterion/redirect.h>
 
-Test(clear_map, should_return_success)
+static void redirect_all_std(void)
```
Comment (HugoN1) [inline server/tests/map_handling/clear_map.c:13]: you know what to do

File: server/tests/map_handling/get_number_ressources_needed.c [hunk 9]
```diff
@@ -10,7 +10,13 @@
 #include <criterion/redirect.h>
 #include <stdlib.h>
 
-Test(get_number_ressources_needed, food_example)
+static void redirect_all_std(void)
```
Comment (HugoN1) [inline server/tests/map_handling/get_number_ressources_needed.c:13]: another time

File: server/tests/map_handling/initialize_map.c [hunk 10]
```diff
@@ -10,7 +10,13 @@
 #include <criterion/criterion.h>
 #include <criterion/redirect.h>
 
-Test(initialize_map, should_return_success)
+static void redirect_all_std(void)
```
Comment (HugoN1) [inline server/tests/map_handling/initialize_map.c:13]: you're close to the end

File: server/tests/server_handling/check_teams_name_exist.c [hunk 11]
```diff
@@ -11,7 +11,13 @@
 #include <criterion/criterion.h>
 #include <criterion/redirect.h>
 
-Test(check_teams_name_exist, should_return_an_index)
+static void redirect_all_std(void)
```
Comment (HugoN1) [inline server/tests/server_handling/check_teams_name_exist.c:14]: still same

File: server/tests/server_handling/connection/clear_logged_clients.c [hunk 12]
```diff
@@ -9,7 +9,13 @@
 #include <criterion/criterion.h>
 #include <criterion/redirect.h>
 
-Test(clear_logged_clients, should_clear_logged_clients)
+static void redirect_all_std(void)
```
Comment (HugoN1) [inline server/tests/server_handling/connection/clear_logged_clients.c:12]: again and again

File: server/tests/server_handling/restore_request_buffer.c [hunk 13]
```diff
@@ -10,7 +10,13 @@
 #include <criterion/criterion.h>
 #include <criterion/redirect.h>
 
-Test(restore_request_buffer, restore_request_buffer)
+static void redirect_all_std(void)
```
Comment (HugoN1) [inline server/tests/server_handling/restore_request_buffer.c:13]: one more effort

File: server/tests/timeout_handling/get_lowest_requested_time.c [hunk 14]
```diff
@@ -18,7 +18,7 @@ static void redirect_all_std(void)
     cr_redirect_stderr();
 }
 
-Test(get_lowest_requested_time, test1)
+Test(get_lowest_requested_time, test1, .init = redirect_all_std)
```
Comment (HugoN1) [inline server/tests/timeout_handling/get_lowest_requested_time.c:21]: LAST ONEEEE

**Review by HugoN1 — CHANGES_REQUESTED** [review]

**Review by HugoN1 — APPROVED** [review]

**Pull request merged**